File: ios/Sources/Application/KnowledgeAnswerSanitizer.swift
import Foundation

/// Output hygiene for on-device generation. Strips leaked runtime control tokens only.
/// Does not rewrite facts, move citations, or reshape prose — keeps the body model-owned.
public enum KnowledgeAnswerSanitizer {
    private static let controlTokenPattern = try! NSRegularExpression(
        pattern: #"</?(?:bos|eos|pad|unk|s|start_of_turn|end_of_turn)\s*>"#,
        options: [.caseInsensitive]
    )

    public static func stripControlTokens(_ text: String) -> String {
        let ns = text as NSString
        let full = NSRange(location: 0, length: ns.length)
        return controlTokenPattern.stringByReplacingMatches(
            in: text,
            options: [],
            range: full,
            withTemplate: ""
        )
    }
}

File: android/application/src/main/kotlin/ai/orynode/mobile/application/KnowledgeAnswerSanitizer.kt
package ai.orynode.mobile.application

object KnowledgeAnswerSanitizer {
    private val controlToken =
        Regex("</?(?:bos|eos|pad|unk|s|start_of_turn|end_of_turn)\\s*>", RegexOption.IGNORE_CASE)

    fun stripControlTokens(text: String): String = controlToken.replace(text, "")
}
